### ISSUE TITLE
cast should(?) accept `Union[str, Type[_T]]`

### ISSUE THREAD

**Issue by ikonst** [python/typeshed#2407]
`cast` is defined as:
```py
def cast(tp: Type[_T], obj: Any) -> _T: ...
```
but it's possible to do `cast('Foo', foo)` when you want to avoid recursive imports. Does this mean that the signature is actually:
```py
def cast(tp: Union[str, Type[_T]], obj: Any) -> _T: ...
```
?

I'd think that since it's not actually a string, we'd need to `NewType` it.

**Comment by ikonst** [comment]
Related: https://youtrack.jetbrains.com/issue/PY-31290

**Comment by JelleZijlstra** [comment]
That generally makes sense to me. I don't think we should use a NewType here; that would require us to write `cast(SomeNewType('Foo'), foo)`.

It might be better to write this as an overload:
```
@overload
def cast(tp: Type[_T], obj: Any) -> _T: ...
@overload
def cast(tp: str, obj: Any) -> Any: ...
```

**Comment by ikonst** [comment]
https://github.com/python/typeshed/pull/2408

**Comment by JelleZijlstra** [comment]
Fixed by #2408.
